deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1PodHandler.java

```diff
@@ -372,4 +385,146 @@ public boolean cleanUp() {
 	public int getWeight() {
 	    return ResourceKind.POD.getWeight();
 	}
+	
+	public void watch(ApiClient apiClient, String appName, String serviceName, String namespace)
+			throws IOException, ApiException, HyscaleException {
+		 OkHttpClient httpClient=getUpdatedHttpClient(apiClient.getHttpClient());
+		 apiClient.setHttpClient(httpClient);
+		 CoreV1Api api = new CoreV1Api(apiClient);
+
+		String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
+		PodParent podParent=getPodParent(apiClient, serviceName, namespace);
+
+		Integer replicas = null;
+		String latestPodSelector = null;
+		V1StatefulSet v1StatefulSet=null;
+		V1StatefulSetHandler v1StatefulSetHandler=null;
+
+		if (podParent.getKind().equals(ResourceKind.STATEFUL_SET.getKind())) {
+			v1StatefulSetHandler = (V1StatefulSetHandler) ResourceHandlers
+					.getHandlerOf(ResourceKind.STATEFUL_SET.getKind());
+			v1StatefulSet = (V1StatefulSet) podParent.getPod();
+			latestPodSelector = v1StatefulSetHandler.getPodSelector(v1StatefulSet, selector);
+			replicas = v1StatefulSet != null ? v1StatefulSet.getSpec().getReplicas() : 1;
+		} else if (podParent.getKind().equals(ResourceKind.DEPLOYMENT.getKind())) {
+			V1DeploymentHandler v1DeploymentHandler = (V1DeploymentHandler) ResourceHandlers
+					.getHandlerOf(ResourceKind.DEPLOYMENT.getKind());
+			
+			V1Deployment v1Deployment = (V1Deployment) podParent.getPod();
+			latestPodSelector = v1DeploymentHandler.getPodSelector(apiClient, selector, true, namespace);
+			replicas = v1Deployment != null ? v1Deployment.getSpec().getReplicas() : 1;
+		}
+		Watch<V1Pod> watch = null;
+		String namespaceSelector=ResourceSelectorUtil.getNamespaceSelector(namespace); 
+
+		Set<String> containersReadyPods = new HashSet<String>();
+		Set<String> readyPods = new HashSet<String>();
+		Set<String> initializedPods = new HashSet<String>();
+		Set<String> createdPods = new HashSet<String>();
+		
+		ActivityContext initializedActivityContext = null;
+		ActivityContext creationActivityContext = null;
+		ActivityContext readyActivityContext = null;
+
+
+		watch = Watch.createWatch(apiClient,
+				api.listPodForAllNamespacesCall(null, null,namespaceSelector , latestPodSelector, null, null, null, replicas*10, Boolean.TRUE, null),
```

Comment:
instead of listPodForAllNamespacesCall, we can use listNamespacedPodCall, then we can remove the namespaceSelector logic